src/geometry/mckinney_gammie_ryan.hpp

```diff
@@ -66,7 +72,10 @@ class McKinneyGammieRyan {
     C[1] = th;
     C[2] = X3;
 
-    const Real drdX1 = std::exp(X1);
+    // const Real drdX1 = std::exp(X1);
+    const Real hexp_dr = X1 - hexp_br_;
+    const Real drdX1 =
+        std::exp(X1 + (hexp_dr > 0.) * hexp_csq_ * std::pow(hexp_dr, hexp_nsq_));
```

Comment:
nice use of masking. I was worried there'd be an if statement.

Reply:
Is drdX1 supposed to be the derivative?  If so this is missing a term, right?  (vs [here](https://github.com/AFD-Illinois/kharma/blob/feature/multizone-dev/kharma/coordinates/coordinate_systems.hpp#L476))

Incidentally, if Phoebus uses numerical first derivatives/transformation matrices, why?

Reply:
@Yurlungur the adroit use of masking is due to @bprather :) Does this actually save time compared to the if statement? It's still evaluating a conditional right? I could be wrong, I'm just curious.

@bprather Yes you're right I dropped a term, thanks for catching that. I'll fix. I'm not sure I understand the numerical transformation matrix concern, could you elaborate?

Reply:
@brryan  the masking vectorize whereas a conditional will not... although it's not clear this vectorizes anyway. (The geometry package as a while probably does not.)

Reply:
It's not a concern, just trying to gauge.  I think in this case, numerical first+second derivatives would actually be more stable, because analytically speaking the derivative of HExp can change really quickly.  Then, a numerical second derivative of the sharp first derivative causes :sparkles: problems :sparkles:

In particular, d/dx of exp(x + pow(x,n)) is exp(x + pow(x, n))*(1 + n*exp(x + pow(x, n-1)), yeah? (Or in any case involves pow(x, n-1), I'm dashing this off).  However, if n=1, pow(x,0) is constant for *all* x, i.e., including x=0.  Thus, the derivative becomes straight-up mathematically discontinuous at x=x_br.  for n~=1, it's still pretty bad from a numerical-derivative perspective, and can lead to some gnarly connection coefficients.  And even for n=2 (which starts to *really* stretch the outer edge) it seems to have a noticeable effect, stemming I think from not preserving conn^mu_mu_nu = d/dx^nu gdet

Reply:
as a general rule, Phoebus supports analytic *and* numerical derivatives. Likely when this code is run, the derivatives will be numerical if the code is run with cached geometry.